src/server/protocol_client.cc

```diff
@@ -314,6 +314,67 @@ io::Result<ProtocolClient::ReadRespRes> ProtocolClient::ReadRespReply(uint32_t t
   return res;
 }
 
+io::Result<TakeRespExpr::Vec> ProtocolClient::TakeRespReply(uint32_t timeout) {
+  auto prev_timeout = sock_->timeout();
+  sock_->set_timeout(timeout);
+  auto res = TakeRespReply(nullptr, false);
+  sock_->set_timeout(prev_timeout);
+  return res;
+}
+
+io::Result<TakeRespExpr::Vec> ProtocolClient::TakeRespReply(base::IoBuf* buffer, bool copy_msg) {
```

Comment:
its the same as ReadRespExpr except some small details. If this code is not temporary, please keep it as one function. Futhermore, if you extend RespExpr to be owning, you can pass it just as a parameter without any templates